### PR TITLE
[WIP] New functionality added

### PR DESCRIPTION
**Pull request by samyakbvs**
I have added to tiles and a functionality that in order to pass one tile the player would need a tool (boat) which he will get in the other tile

I made this change to make the game more interesting

Deploy link - http://rawgit.com/samyakbvs/labyrinth/master/index.html

### CODE REVIEW

**Comment by samyakbvs** [comment]
@niccokunzmann

File: js/tiles.js [hunk 1]
```diff
@@ -379,6 +379,54 @@ const door = {
...
+          this.wallTop = this.createImage("tiles/rooms/wall/top.svg");
+          this.wallRight = this.createImage("tiles/rooms/wall/right.svg");
+          this.ground = this.createImage("tiles/rooms/floor/river.svg");
+      },
+      visit: function() {
+          if(player.inventory.has('Boat')){
+              swal({
+            type: 'success',
+            title: 'You got the treasure !',
+            text: "there's more to find !"
```
Comment (abishekvashok) [inline js/tiles.js]: pls intend this properly

**Comment by arjunpandey2003** [comment]
@samyakbvs pls send a screenshot of your tile and where it is placed I cant find it nor did I get any alert I checked the whole labyrinth.

**Comment by samyakbvs** [comment]
Actually, I have done my part, the rest of the things will be completed by YashKumarVerma as we are doing pair programming

**Comment by samyakbvs** [comment]
He said he will do his part till 12, so guys pls wait

File: js/tiles.js [hunk 2]
```diff
@@ -379,6 +379,54 @@ const door = {
       this.ground.show();
     },
   }),
+  river: Object.assign({}, OpenDoors, {
+      canEnterFromTheRight() {return false;},
+      canLeaveToTheRight() {return false;},
+      canEnterFromTheLeft() {return false;},
+      canLeaveToTheLeft() {return false;},
```
Comment (niccokunzmann) [inline js/tiles.js]: If the player cannot enter, how can the player visit the tile?
I think, the first argument to this function is always "player" although it is not described here. JavaScript allows not using arguments. 

You could do something like: 
```
canEnterFromTheRight() {
return player.inventory.has('Boat')
} 

```

File: js/tiles.js [hunk 3]
```diff
@@ -379,6 +379,54 @@ const door = {
...
+  boat: Object.assign({}, OpenDoors, {
+	createImages: function() {
+      this.ground = this.createImage("tiles/rooms/floor/boat.svg");
+      this.wallTop = this.createImage("tiles/rooms/wall/top.svg");
+      this.wallRight = this.createImage("tiles/rooms/door/right.svg");
+    },
+    visit: function() {
+          swal({
+	        type: 'info',
+	        title: 'You found a boat !',
```
Comment (niccokunzmann) [inline js/tiles.js]: nice :)

File: js/tiles.js [hunk 4]
```diff
@@ -379,6 +379,54 @@ const door = {
...
+	        text: "Keep it it might come handy"
+	      });
+          // add item to inventory
+          player.inventory.add(['Boat','way.png']);
```
Comment (niccokunzmann) [inline js/tiles.js]: I think, the boat could be in a "boat.png" could you rename this?

**Comment by niccokunzmann** [comment]
@samyakbvs I added [WIP] into the title. It means work in progress. You can remove it when the pull request is ready to merge.

**Comment by samyakbvs** [comment]
Okay

File: js/tiles.js (hunk 3 above)
Comment (samyakbvs) [reply js/tiles.js]: Thanks

**Comment by samyakbvs** [comment]
@niccokunzmann pls merge

File: js/tiles.js [hunk 5]
```diff
@@ -402,6 +402,54 @@ const door = {
       this.ground.show();
     },
   }),
+  river: Object.assign({}, OpenDoors, {
+      canEnterFromTheRight() {return player.inventory.has('Boat');},
```
Comment (niccokunzmann) [inline js/tiles.js]: This should be an error: player is not defined.

Comment (niccokunzmann) [reply js/tiles.js]: Please do not use the global variable.

**Comment by niccokunzmann** [comment]
Thanks!

**Pull request merged**